src/tools/kotlin-class-generator.ts

```diff
@@ -277,33 +287,32 @@ ${lines}
     return '\n' + ktUtils.indent(blocks.join('\n'), 2);
   }
 
-  generateClasses(schemaHash: string): string {
-    const fieldCount = Object.keys(this.node.schema.fields).length;
-    const withFields = (populate: string) => fieldCount === 0 ? '' : populate;
-
-    return `\
-
-    ${this.generateClassDefinition()} {
-${this.generateFieldsDefinitions()}
-        ${this.generateCopyMethods()}
-    ${this.opts.wasm ? `
+  private maybeGenerateWasmSpecificMethods() {
+    if (!this.opts.wasm) return '';
+    return `
         fun reset() {
-          ${withFields(`${this.fieldsReset.join('\n            ')}`)}
+            ${ktUtils.indentFollowing(this.fieldsReset, 3)}
         }
 
         override fun encodeEntity(): NullTermByteArray {
             val encoder = StringEncoder()
             encoder.encode("", entityId)
-            ${this.encode.join('\n        ')}
+            ${ktUtils.indentFollowing(this.encode, 3)}
             return encoder.toNullTermByteArray()
         }
 
         override fun toString() =
             "${this.className}(${this.fieldsForToString.join(', ')})"
-    ` : ''}
-        companion object : ${this.prefixTypeForRuntime('EntitySpec')}<${this.className}> {
+`;
+  }
+
+  private generateEntitySpec() {
+    const fieldCount = Object.keys(this.node.schema.fields).length;
+    const withFields = (populate: string) => fieldCount === 0 ? '' : populate;
```

Comment:
Ideally, it would be good to replace `withFields` IMO. WDYT?

Reply:
I replaced / inlined / simplified a bit :) Thanks for the suggestion.